### ISSUE TITLE
Data process for Greek nouns

### ISSUE THREAD

**Issue by andrewtavis** [scribe-org/Scribe-Data#58]
### Terms

- [X] I have searched [open and closed data issues](https://github.com/scribe-org/Scribe-Data/issues?q=is%3Aissue+label%3Adata+)
- [X] I agree to follow Scribe-Data's [Code of Conduct](https://github.com/scribe-org/Scribe-Data/blob/main/.github/CODE_OF_CONDUCT.md)

### Languages

Greek

### Description

This issue would create the base noun and verb SPARQL queries for Scribe-Data so that we can then add them in to [Scribe-iOS](https://github.com/scribe-org/Scribe-iOS) :) These would be added to the [extract transform](https://github.com/scribe-org/Scribe-Data/tree/main/src/scribe_data/extract_transform) directory.

**Comment by NikosLikomitros** [comment]
Joined from the Greek community.

**Comment by andrewtavis** [comment]
Great to have you, @NikosLikomitros! As discussed in the call, I'll get the base SPARQL queries up and it'd be great if you can check them :)

**Comment by NikosLikomitros** [comment]
Yes, I can help with this. If this helps my language to be further present on the digital editing systems, I can di

**Comment by wkyoshida** [comment]
Breaking this issue into separate issues for the data processes for the Greek nouns and verbs.

This will become the issue for the SPARQL query for Greek nouns.

The [`WIKIDATAGUIDE.md`](https://github.com/scribe-org/Organization/blob/main/WIKIDATAGUIDE.md) has some good info on Wikidata and SPARQL :blush: 

As reference, this is the query used for the Italian nouns [`.../Italian/nouns/query_nouns.sparql`](https://github.com/scribe-org/Scribe-Data/blob/main/src/scribe_data/extract_transform/Italian/nouns/query_nouns.sparql)

**Comment by Jk40git** [comment]
> Breaking this issue into separate issues for the data processes for the Greek nouns and verbs.
> 
> This will become the issue for the SPARQL query for Greek nouns.
> 
> The [`WIKIDATAGUIDE.md`](https://github.com/scribe-org/Organization/blob/main/WIKIDATAGUIDE.md) has some good info on Wikidata and SPARQL 😊
> 
> As reference, this is the query used for the Italian nouns [`.../Italian/nouns/query_nouns.sparql`](https://github.com/scribe-org/Scribe-Data/blob/main/src/scribe_data/extract_transform/Italian/nouns/query_nouns.sparql)

@wkyoshida the link for the reference is broken please

**Comment by wkyoshida** [comment]
Hey @Jk40git :wave: 
There was a recent change to put all the language processing files under a parent `languages/` directory, so that file's location just moved a bit to here:
- https://github.com/scribe-org/Scribe-Data/blob/main/src/scribe_data/extract_transform/languages/Italian/nouns/query_nouns.sparql

**Comment by Jk40git** [comment]
> Hey @Jk40git 👋 There was a recent change to put all the language processing files under a parent `languages/` directory, so that file's location just moved a bit to here:
> 
> * https://github.com/scribe-org/Scribe-Data/blob/main/src/scribe_data/extract_transform/languages/Italian/nouns/query_nouns.sparql

I am not getting any result with the greek nouns no matching record found

**Comment by wkyoshida** [comment]
Hey @Jk40git!
There definitely are Greek nouns on Wikidata, so it could just be something with the query. Would you like to share what you tried in a [gist](https://gist.github.com/) and we could take a look perhaps?

I would also make sure to go over the [`WIKIDATAGUIDE.md`](https://github.com/scribe-org/Organization/blob/main/WIKIDATAGUIDE.md) if you haven't yet. Wikidata queries can be tough to work with :sweat_smile:

**Comment by andrewtavis** [comment]
Hey there @Jk40git :) Makes sense that this was a bit tough. Specifically you were using `Q9129` in the query for `Greek`, but then there are multiple kinds of Greek languages including `Q36510` (Modern Greek) and `Q35497` (Ancient Greek). 

Here's the correct way of going about this using `Q36510`:

```sql
SELECT DISTINCT ?singular ?plural ?gender WHERE {

  # Nouns and pronouns.
  VALUES ?nounTypes { wd:Q1084 wd:Q147276 }
  ?lexeme a ontolex:LexicalEntry ;
    dct:language wd:Q36510;
    wikibase:lexicalCategory ?noun .
  FILTER(?noun = ?nounTypes)

  # Optional selection of singular forms.
  OPTIONAL {
    ?lexeme ontolex:lexicalForm ?singularForm .
    ?singularForm ontolex:representation ?singular ;
      wikibase:grammaticalFeature wd:Q131105 ;
      wikibase:grammaticalFeature wd:Q110786 ;
  } .

  # Optional selection of plural forms.
  OPTIONAL {
    ?lexeme ontolex:lexicalForm ?pluralForm .
    ?pluralForm ontolex:representation ?plural ;
      wikibase:grammaticalFeature wd:Q131105 ;
      wikibase:grammaticalFeature wd:Q146786 ;
  } .

  # Optional selection of genders.
  OPTIONAL {
    ?lexeme wdt:P5185 ?nounGender .
    FILTER NOT EXISTS { ?lexeme wdt:P31 wd:Q48277}
  } .

  SERVICE wikibase:label {
    bd:serviceParam wikibase:language "[AUTO_LANGUAGE]".
    ?nounGender rdfs:label ?gender .
  }
}
```

**Comment by Jk40git** [comment]
> Hey there @Jk40git :) Makes sense that this was a bit tough. Specifically you were using `Q9129` in the query for `Greek`, but then there are multiple kinds of Greek languages including `Q36510` (Modern Greek) and `Q35497` (Ancient Greek).
> 
> Here's the correct way of going about this using `Q36510`:
> 
> ```sql
> SELECT DISTINCT ?singular ?plural ?gender WHERE {
> 
>   # Nouns and pronouns.
>   VALUES ?nounTypes { wd:Q1084 wd:Q147276 }
>   ?lexeme a ontolex:LexicalEntry ;
>     dct:language wd:Q36510;
>     wikibase:lexicalCategory ?noun .
>   FILTER(?noun = ?nounTypes)
> 
>   # Optional selection of singular forms.
>   OPTIONAL {
>     ?lexeme ontolex:lexicalForm ?singularForm .
>     ?singularForm ontolex:representation ?singular ;
>       wikibase:grammaticalFeature wd:Q131105 ;
>       wikibase:grammaticalFeature wd:Q110786 ;
>   } .
> 
>   # Optional selection of plural forms.
>   OPTIONAL {
>     ?lexeme ontolex:lexicalForm ?pluralForm .
>     ?pluralForm ontolex:representation ?plural ;
>       wikibase:grammaticalFeature wd:Q131105 ;
>       wikibase:grammaticalFeature wd:Q146786 ;
>   } .
> 
>   # Optional selection of genders.
>   OPTIONAL {
>     ?lexeme wdt:P5185 ?nounGender .
>     FILTER NOT EXISTS { ?lexeme wdt:P31 wd:Q48277}
>   } .
> 
>   SERVICE wikibase:label {
>     bd:serviceParam wikibase:language "[AUTO_LANGUAGE]".
>     ?nounGender rdfs:label ?gender .
>   }
> }
> ```

okay thanks for the info. I appreciate!

**Comment by andrewtavis** [comment]
Closed via #93 😊 Thanks for the help here, @Jk40git!
